service/src/lib.rs

```diff
@@ -39,59 +35,50 @@ pub trait Factory: Send + Sync {
     fn get_metadata(&self) -> DeploymentMetadata;
 }
 
-/// Used to get resources of type `T` from factories.
-///
-/// This is mainly meant for consumption by our code generator and should generally not be called by users.
-///
-/// ## Creating your own managed resource
-///
-/// You may want to create your own managed resource by implementing this trait for some builder `B` to construct resource `T`.
-/// [`Factory`] can be used to provision resources on Shuttle's servers if your service will need any.
-///
-/// Please refer to `shuttle-examples/custom-resource` for examples of how to create custom resource. For more advanced provisioning
-/// of custom resources, please [get in touch](https://discord.gg/shuttle) and detail your use case. We'll be interested to see what you
-/// want to provision and how to do it on your behalf on the fly.
-///
-/// ```
+/// Allows implementing plugins for the Shuttle main function.
 #[async_trait]
-pub trait ResourceBuilder<T> {
-    /// The type of resource this creates
-    const TYPE: Type;
+pub trait ResourceBuilder {
+    /// The type of resource this plugin creates.
+    const TYPE: resource::Type;
 
-    /// The internal config being constructed by this builder. This will be used to find cached [Self::Output].
-    type Config: Serialize;
+    /// The input config to this resource.
+    type Config: Default + Serialize;
 
-    /// The output type used to build this resource later
+    /// The output from requesting this resource.
+    /// A cached copy of this will be used if the same [`ResourceBuilder::Config`] is found for this [`ResourceBuilder::TYPE`].
     type Output: Serialize + DeserializeOwned;
 
-    /// Create a new instance of this resource builder
-    fn new() -> Self;
```

Comment:
Why are we dropping `new()`? I originally imagined we would need it in the future to allow resources to be configurable (`local_uri = "..."`)

Reply:
All of those args are builder pattern, and there is never a way to set args in `new`. I instead added `Default` trait bound, since `new` always just called or returned a default value. Did you have other plans for `new`?

Reply:
But isn't `new()` a clearer indication that this is the builder pattern than `default()`? (The `ResourceBuilder` mimics the builder pattern)

Reply:
I see how `default()` can fix my original question though. So am now happy having this either way.

(btw the `Default` trait might be missing on `ResourceBuilder`)